src/languages/latex.js

```diff
@@ -1,70 +1,231 @@
+    return {
+      contains: [SPACE_GOBBLER],
+      starts: {
         relevance: 0,
-        variants: [
-          {begin: /\$\$/, end: /\$\$/},
-          {begin: /\$/, end: /\$/}
-        ]
+        contains: arg,
+        starts: starts_mode
+      }
+    };
+  };
+  var CSNAME = function(csname, starts_mode) {
+    return hljs.inherit(
+      {
+        begin: '\\\\' + csname + '(?![a-zA-Z@:_])',
+        keywords: {$pattern: /\S+/, keyword: '\\' + csname}
       },
-      hljs.COMMENT(
-        '%',
-        '$',
-        {
-          relevance: 0
-        }
-      )
+      ARGUMENT_AND_THEN(ARGUMENT_NONE, starts_mode)
+    );
+  };
+  var BEGIN_ENV = function(envname, starts_mode) {
+    return hljs.inherit(
+      {
+        begin: '\\\\begin(?=\\s*\\r?\\n?\\s*\\{' + envname + '\\})',
+        keywords: {$pattern: /\S+/, keyword: '\\begin'}
+      },
+      ARGUMENT_AND_THEN(ARGUMENT_M, starts_mode)
+    );
+  };
+  var VERBATIM_DELIMITED_EQUAL = hljs.END_SAME_AS_BEGIN({
+    className: 'string',
+    begin: /(.|\r?\n)/, end: /(.|\r?\n)/,
+    excludeBegin: true, excludeEnd: true,
+    endsParent: true
+  });
+  var VERBATIM_DELIMITED_BRACES = {
+    className: 'string',
+    begin: /\{/, end: /\}/,
+    excludeBegin: true, excludeEnd: true,
+    endsParent: true
+  };
+  var VERBATIM_DELIMITED_ENV = function(envname) {
+    return {
+      className: 'string',
+      end: '(?=\\\\end\\{' + envname + '\\})'
+    };
+  };
+  var PATCH_CLASSNAME = function(mode, class_name) {
+    return hljs.inherit(mode, {className: class_name});
+  };
+  var VERBATIM = {
+    variants: [
+      CSNAME('verb', {contains: [VERBATIM_DELIMITED_EQUAL]}),
+      CSNAME('lstinline', {contains: [VERBATIM_DELIMITED_EQUAL]}),
+      CSNAME('mint', ARGUMENT_AND_THEN(ARGUMENT_M, {contains: [VERBATIM_DELIMITED_EQUAL]})),
+      CSNAME('mintinline', ARGUMENT_AND_THEN(ARGUMENT_M, {contains: [VERBATIM_DELIMITED_BRACES, VERBATIM_DELIMITED_EQUAL]})),
+      CSNAME('url', {contains: [PATCH_CLASSNAME(VERBATIM_DELIMITED_BRACES, 'link'),
+                                PATCH_CLASSNAME(VERBATIM_DELIMITED_EQUAL, 'link')]}),
+      BEGIN_ENV('verbatim',    {starts: VERBATIM_DELIMITED_ENV('verbatim')}),
+      BEGIN_ENV('verbatim\\*', {starts: VERBATIM_DELIMITED_ENV('verbatim\\*')}),
+      BEGIN_ENV('Verbatim',     ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('Verbatim')})),
+      BEGIN_ENV('Verbatim\\*',  ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('Verbatim\\*')})),
+      BEGIN_ENV('BVerbatim',    ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('BVerbatim')})),
+      BEGIN_ENV('BVerbatim\\*', ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('BVerbatim\\*')})),
+      BEGIN_ENV('LVerbatim',    ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('LVerbatim')})),
+      BEGIN_ENV('LVerbatim\\*', ARGUMENT_AND_THEN(ARGUMENT_O, {starts: VERBATIM_DELIMITED_ENV('LVerbatim\\*')})),
```

Comment:
Are you double nesting starts here?  `ARGUMENT_AND_THEN` already has `starts` (line 161), I'm not sure why it's also listed here?

Reply:
I am. What I need to do after finding, say, `\begin{minted}` is looking for one optional pair of brackets (with arbitrary highlighted content) followed by exactly one pair of braces (with arbitrary highlighted content) followed by the verbatim content (i.e. `string`). The only possibility for adding such a sequence of syntax elements I saw is `starts` (used repeatedly), but maybe I missed something?

Reply:
You're writing `{starts: { starts: ... }}` though from what I can see you don't need the extra empty starts.

Reply:
Why not:

```
BEGIN_ENV('Verbatim',     ARGUMENT_AND_THEN(ARGUMENT_O, VERBATIM_DELIMITED_ENV('Verbatim'))),
```

Reply:
(Oh, and an arbitrary number of spaces and one newline should be allowed between these elements, but that's a detail.)

Reply:
Ah! That is a good question. No idea, it's an oversight. Will change it.